src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -108,6 +108,26 @@ public void addSession(Name name, Session session) {
         Student student = students.getStudentWithName(name);
         students.addSession(student, session);
     }
+    /**
+     * Removes {@code sessionIndex} of {@code name} from this {@code AddressBook}.
+     * {@code name} and {@sessionIndex} must exist in the address book.
+     */
+    public void removeSession(Name name, Index sessionIndex) {
+        requireAllNonNull(name, sessionIndex);
+        Student student = students.getStudentWithName(name);
+        students.remove(student);
```

Comment:
This `removeSession` method seems to be very destructive. Could this probably be `student.getListOfSessions().remove(sessionIndex)`? 💥 💣

Reply:
No sir, it should be `students.deleteSession(student, sessionIndex);`

Reply:
Roger doger!